Heisenbug in test_helpers

About every five times I run the helpers unit test, compare_dicts fails like this:
```
        d = collate.Compare('long_column_name','=',
            {'really long string key that is similar to others': 'really long string value that is similar to others',
             'really long string key that is like others': 'really long string value that is like others',
             'different key': 'really long string value that is quite alike to others',
             'ni': 'really long string value that is also like everything else'}, [], maxlen=32).quantities
        assert len(d) == 4
        assert len(set(d.values())) == len(d)
        assert len(set(d.keys())) == len(d)
>       assert all(len(k) <= 32 for k in d.keys())
E       assert False
E        +  where False = all(<generator object <genexpr> at 0x7f42fe124558>)
```

This is on Python 3.

Aha, caught it in the debugger:

```
>       assert all(len(k) <= 32 for k in d.keys())
E       assert False
E        +  where False = all(<generator object test_compare_lists.<locals>.<genexpr> at 0x10b6289e8>)

tests/test_helpers.py:43: AssertionError
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> entering PDB >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
> /Users/mbauman/dssg/collate/tests/test_helpers.py(43)test_compare_lists()
-> assert all(len(k) <= 32 for k in d.keys())
(Pdb) p d
{'really_long_column_name_=_rea_01': ("(really_long_column_name = 'really long string value that is similar to others')::INT",), 'really_long_column_name_=_rea_04': ("(really_long_column_name = 'really long string value that is quite alike to others')::INT",), 'really_long_column_name_=_rea_03': ("(really_long_column_name = 'really long string value that is also like everything else')::INT",), 'really_long_column_name_=_really long string value that is like others': ("(really_long_column_name = 'really long string value that is like others')::INT",)}
(Pdb) p d.keys()
dict_keys(['really_long_column_name_=_rea_01', 'really_long_column_name_=_rea_04', 'really_long_column_name_=_rea_03', 'really_long_column_name_=_really long string value that is like others'])
(Pdb) p [len(k) for k in d.keys()]
[32, 32, 32, 70]
```

--- a/collate/collate.py
+++ b/collate/collate.py
@@ -262,7 +262,7 @@ class Compare(Aggregate):
         if include_null:
             d['{}_{}'.format(col, include_null)] = '({} is NULL)::INT'.format(col)
         if maxlen is not None and any(len(k) > maxlen for k in d.keys()):
-            for i, k in enumerate(d.keys()):
+            for i, k in enumerate(list(d.keys())):
                 d['%s_%02d' % (k[:maxlen-3], i)] = d.pop(k)
 
         Aggregate.__init__(self, d, function, order)

